test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTextAreaTest.cs

```diff
@@ -40,6 +42,73 @@ public void TextAreaFor_DoesNotGeneratePlaceholderAttribute_WhenNoPlaceholderPre
             Assert.DoesNotContain(@"placeholder=""HtmlEncode[[placeholder]]""", result, StringComparison.Ordinal);
         }
 
+        public static TheoryData TextAreaFor_UsesModelStateValueForComplexExpressionsData
+        {
+            get
+            {
+                return new TheoryData<Expression<Func<ComplexModel, string>>, string>
+                {
+                    {
+                        model => model.Property3["key"],
+                        "<textarea id=\"HtmlEncode[[pre_Property3_key_]]\" name=\"HtmlEncode[[pre.Property3[key]]]\">" + Environment.NewLine +
+                        "HtmlEncode[[Prop3Val]]</textarea>"
+                    },
+                    {
+                        model => model.Property4.Property5,
+                        "<textarea id=\"HtmlEncode[[pre_Property4_Property5]]\" name=\"HtmlEncode[[pre.Property4.Property5]]\">" + Environment.NewLine +
+                        "HtmlEncode[[Prop5Val]]</textarea>"
+                    },
+                    {
+                        model => model.Property4.Property6[0],
+                        "<textarea id=\"HtmlEncode[[pre_Property4_Property6_0_]]\" name=\"HtmlEncode[[pre.Property4.Property6[0]]]\">" + Environment.NewLine +
+                        "HtmlEncode[[Prop6Val]]</textarea>"
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TextAreaFor_UsesModelStateValueForComplexExpressionsData))]
+        public void TextAreaFor_ComplexExpressions_SetsValueFromViewData(
```

Comment:
Rename test or member data to align w/ what this is actually testing: Shows that the expression value is used when generating the body of the `<textarea>` element.

However, this is already covered in `HtmlHelperTextAreaExtensionsTest`. Should instead test use of values from `ModelState` and `ViewData[x]`.